design/object-bucket/object-bucket.md

```diff
@@ -0,0 +1,300 @@
+# Rook-Ceph Bucket Provisioning
+
+## Proposed Feature
+
+Enable Rook-Ceph Object users the ability to request buckets in an object store through the ObjectBucketClaim (OBC) and ObjectBucket (OB) API resource types.
+
+### Overview
+
+Rook-Ceph has developed Custom Resource Definitions (CRDs) to support the dynamic provisioning of object stores and users (CephObjectStores and CephObjectStoreUsers, respectively).  This proposal introduces the next logical step in this design: the addition a controller for handling dynamic bucket creation requests via the generalized ObjectBucketClaim and ObjectBucket APIs.  These APIs are defined in an external project with the intent of working to garner broader community support.  The intent is to round out the Rook-Ceph experience for Kubernetes users by providing a single control plane for the managing of Ceph Object components.  The ObjectBucketClaim and ObjectBucket APIs represent generic S3 bucket creation requests to enable the provisioning of buckets by most object store providers.
```

Comment:
So the CRDs will be defined in the external repo? Having the external design doc would really help me understand. If that's the case, I'm not clear why the OBC and OB types are defined in the `rook.io/v1` group. If they are defined in `rook.io/v1`, it seems like there is nothing left to be defined externally.

Reply:
Apologies for the confusion, the `rook.io/v1` is an erroneous left over.  The group/version should just read `bucket/v1alpha1` or similar.